astropy/modeling/core.py

```diff
@@ -1961,14 +1965,15 @@ class members that should be added to the generated
         ``standard_broadcasting``, and ``__module__`.  This is currently for
         internal use only.
         """
-
         # Note, currently this only supports binary operators, but could be
         # easily extended to support unary operators (namely '-') if/when
         # needed
         children = []
         for child in (left, right):
             if isinstance(child, (_CompoundModelMeta, _CompoundModel)):
-                children.append(child._tree)
+                children.append(copy.deepcopy(child._tree))
```

Comment:
It's basically the same argument as with single models - if it's a reference, any tweaks to it can lead to unexpected results. A real example of this is a time-dependent distortion. The same distoriton model applies to different observations but needs to be tweaked based on the time the observation was taken. It can be left to the user to do the copy. OTH it can be easily overlooked in the documentation and it can be hard to see there's unexpected difference in the result. I'll add a comment in case we need to revise this in the future.